en/02_Developer_Guides/03_Forms/07_Using_GridField_With_Arbitrary_Data.md

```diff
@@ -0,0 +1,437 @@
+    {
+        return BasicSearchContext::create(static::class);
+    }
+
+    public function scaffoldSearchFields()
+    {
+        return FieldList::create([
+            HiddenField::create(BasicSearchContext::config()->get('general_search_field_name')),
+            TextField::create('Title', 'Title'),
+        ]);
+    }
+
+    // ...
+}
+```
+
+No changes are required to the `GridField` components, assuming you didn't remove the `GridFieldFilterHeader` component.
+
+[hint]
+The `BasicSearchContext` respects some (*but not all*) [`$searchable_fields` configuration options](/developer_guides/model/scaffolding/#searchable-fields), so you can implement a `searchableFields()` method in your class to further customise the `GridField` filtering experience.
+[/hint]
+
+### Exporting data {#custom-export}
+
+Just like with `ArrayData`, to export or print data we don't need to do anything more than ensure the relevant components are in the `GridField` config.
+
+```php
+use SilverStripe\Forms\GridField\GridFieldExportButton;
+use SilverStripe\Forms\GridField\GridFieldPrintButton;
+$gridField->getConfig()->addComponents([
+    GridFieldExportButton::create('buttons-before-left'),
+    GridFieldPrintButton::create('buttons-before-left'),
+]);
+```
+
+### Viewing data in a form {#custom-view}
+
+The same requirement of a positive integer `ID` field as described in [viewing `ArrayData` in a form](#arraydata-view) above applies here too.
+
+If the class representing your data has a `getCMSFields()` method, the return value of that method will be used for the fields displayed in form.
+
+If your class doesn't implement a `canEdit()` method, or it does and the method returns `false`, the form will be read-only.
+
+[hint]
+You can optionally implement the `i18n_plural_name()` method to return a localised string to represent the singular name of this model. This is used in the add button, breadcrumbs, and toasts.
+[/hint]
+
+```php
+namespace App\Data;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\HiddenField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\View\ViewableData;
+
+class DataRepresentation extends ViewableData
+{
+    // ...
+
+    public function getCMSFields()
+    {
+        return FieldList::create([
+            HiddenField::create('ID'),
+            TextField::create('FeldName', 'View Field Label'),
+        ]);
+    }
+
+    // ...
+}
+```
+
+You will need to have `GridFieldDetailForm` and `GridFieldViewButton` components in your `GridField` config in order to access the form view.
+
+```php
+use SilverStripe\Forms\GridField\GridFieldDetailForm;
+use SilverStripe\Forms\GridField\GridFieldViewButton;
+
+$gridField->getConfig()->addComponents([
+    GridFieldViewButton::create(),
+    GridFieldDetailForm::create(),
+]);
+```
+
+### Editing data in a form {#custom-edit}
+
+To edit data, there are a few extra pre-requisites - but this is where representing the data in your own class gets really powerful.
```

Comment:
The second part of the sentence doesn't convey any information.

```suggestion
There are a few extra pre-requisites to allow content authors to edit data.
```

Reply:
It conveys the information that this is a more powerful option than just using `ArrayData` - that's not nothing. I'm going to leave it in unless you have a more convincing reason to remove it.

Reply:
I've made the suggested change to the first part of the sentence.

Reply:
Humm ... I'm thinking if we want to make the case for why you would want to use your own class, we should make it directly after the "Representing data in your own class" header ... rather than in the 5th sub-sub-section.

Maybe something like:

> Representing data in your own class adds some complexity, but empowers your content authors to create, update and delete entries via the GridField.

Reply:
Done and removed the second part of that sentence, even though I still like it :p